src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1274 @@
+module SoilBiogeochemDecompCascadeMIMICSMod
+
+  !-----------------------------------------------------------------------
+  ! !DESCRIPTION:
+  ! Sets the coeffiecients used in the decomposition cascade submodel.  
+  ! This uses the MIMICS parameters.
+  !
+  ! !USES:
+  use shr_kind_mod                       , only : r8 => shr_kind_r8
+  use shr_const_mod                      , only : SHR_CONST_TKFRZ
+  use shr_log_mod                        , only : errMsg => shr_log_errMsg
+  use clm_varpar                         , only : nlevdecomp, ndecomp_pools_max
+  use clm_varpar                         , only : i_met_lit, i_cop_mic, i_oli_mic, i_cwd
+  use clm_varpar                         , only : i_litr_min, i_litr_max, i_cwdl2
+  use clm_varctl                         , only : iulog, spinup_state, anoxia, use_lch4, use_fates
+  use clm_varcon                         , only : zsoi
+  use decompMod                          , only : bounds_type
+  use spmdMod                            , only : masterproc
+  use abortutils                         , only : endrun
+  use CNSharedParamsMod                  , only : CNParamsShareInst, nlev_soildecomp_standard 
+  use SoilBiogeochemDecompCascadeConType , only : decomp_cascade_con
+  use SoilBiogeochemStateType            , only : soilbiogeochem_state_type
+  use SoilBiogeochemCarbonFluxType       , only : soilbiogeochem_carbonflux_type
+  use SoilBiogeochemCarbonStateType      , only : soilbiogeochem_carbonstate_type
+  use SoilStateType                      , only : soilstate_type
+  use TemperatureType                    , only : temperature_type 
+  use CNVegCarbonFluxType                , only : cnveg_carbonflux_type
+  use ch4Mod                             , only : ch4_type
+  use ColumnType                         , only : col                
+  use GridcellType                       , only : grc
+  use SoilBiogeochemStateType            , only : get_spinup_latitude_term
+
+  !
+  implicit none
+  private
+  !
+  ! !PUBLIC MEMBER FUNCTIONS:
+  public :: readParams                      ! Read in parameters from params file
+  public :: init_decompcascade_mimics       ! Initialization
+  public :: decomp_rates_mimics             ! Figure out decomposition rates
+  !
+  ! !PUBLIC DATA MEMBERS 
+  !
+  ! !PRIVATE DATA MEMBERS 
```

Comment:
I think more of the variables should have a description for them. Some of them have a structure with numbers to indicate levels and pools and such, so you just need to describe how the structure of the name works.

Reply:
I added comments. Possibly more could be added still.